### DISCUSSION TITLE
How can I help on this?

### DISCUSSION THREAD

**Discussion by fredguth** [sl-nx/superlogin-next#47] [General]
https://github.com/perfood/couch-auth/projects/1#card-69052632

I would like to remove the express dependency.  I plan to use couch-auth in a sveltekit app.  
Could an issue be created with all the steps needed to be developed?

**Comment by fynnlyte** [comment]
So far this is just an idea, but if you want to give it a shot, I'd probably do it like this:

1. Extract everything that happens in `user.ts` + its dependencies (`dbAuth`, `mailer`,...) into a separate `couch-auth-core` module. Most of the programmatic API could be used without `express`.
2. Add this module as dependency to `couch-auth`, import it in `index.ts` and `routes.ts` to get the current functionality.

But I'm pretty sure that the devil will be in the detail...

**Reply by fredguth** [reply]
Thanks for the blueprint.
